ext/pdo_firebird/firebird_statement.c

```diff
@@ -429,7 +460,7 @@ static int firebird_stmt_get_col(
 					break;
 				case SQL_DOUBLE:
 					/* TODO: Why is this not returned as the native type? */
-					ZVAL_STR(result, zend_strpprintf(0, "%F", *(double*)var->sqldata));
+					ZVAL_STR(result, zend_strpprintf(0, "%F", get_double_from_sqldata(var->sqldata)));
```

Comment:
Shouldn't something similar for ``SQL_FLOAT`` be done? as I imagine the same issue is present

Reply:
A float is 4 bytes, and the minimum alignment on every reasonable 32-bit platform is at least 4, so that can't be an issue.

Reply:
> Could the test try and use every single SQL Type that we currently support?

Hmm sure, I guess it doesn't hurt although it'll take me some time to amend the test :P

Reply:
> A float is 4 bytes, and the minimum alignment on every reasonable 32-bit platform is at least 4, so that can't be an issue.

Ah right, yeah that's fair